Header header
bool trigger
string state